eodag/api/core.py

```diff
@@ -185,6 +185,7 @@ def __init__(
                         os.path.join(self.conf_dir, "shp"),
                     )
         self.set_locations_conf(locations_conf_path)
+        self.download_info = {}
```

Comment:
Please store full SearchResult objects in a `_search_results_cache = collections.dequeue()` having `maxlen` default to 100 (stored in a new constant) and customizable using an environment variable like `EODAG_SEARCH_RESULTS_CACHE_MAXLEN`.

Please also put `_search_results_cache` in `rest.utils` as it only concerns server-mode downloads.

To find a matching product, loop over the cache starting with the most recent result. (check at first only provider and product type of the first product of each search result)